File: 2021/01/v2.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func main() {
	part1()
	part2()
}

func part1() { solve(1) } // 1564

func part2() { solve(3) } // 1611

func solve(windowSize int) {
	// filename := "example.txt"
	filename := "input.txt"
	bytes, _ := ioutil.ReadFile(filename)
	lines := strings.Split(string(bytes), "\n")
	numbers := Atoi(lines)

	c := 0
	lastSum := 0

	for i := range numbers {
		if i < windowSize-1 {
			continue
		}

		currSum := 0
		for j := 0; j < windowSize; j++ {
			currSum += numbers[i-j]
		}

		if currSum > lastSum && lastSum > 0 {
			c++
		}

		lastSum = currSum
	}

	fmt.Println(c)
}

func Atoi(lines []string) []int {
	numbers := make([]int, len(lines))
	for i, l := range lines {
		n, _ := strconv.Atoi(l)
		numbers[i] = n
	}
	return numbers
}

File: 2021/07/v2.ts
import {
  compose,
  identity,
  map,
  range,
  split,
  subtract,
  sum,
} from "https://raw.githubusercontent.com/selfrefactor/rambda/master/dist/rambda.esm.js";
import * as R from "https://raw.githubusercontent.com/selfrefactor/rambda/master/dist/rambda.esm.js";
import { green } from "https://deno.land/std@0.117.0/fmt/colors.ts";

const printSolution = (part, answer) => console.info(green(part), "\t", answer);

const useExample = 0;
const filename = useExample ? "example.txt" : "input.txt";
const data = Deno.readTextFileSync(filename);
const positions = compose(
  map(parseInt),
  split(","),
)(
  data,
) as number[];

const abs = Math.abs;
const min = Math.min;
const max = Math.max;

const totalFuel = (fuelConsumption) =>
  (positions) =>
    (stop) =>
      compose(
        sum,
        map(fuelConsumption),
        map(abs),
        map(subtract(stop)),
      )(positions);

const possiblePositions = range(min(...positions), max(...positions) + 1);

const minFuel = (fuelConsumption) =>
  min(...map(
    totalFuel(fuelConsumption)(positions),
    possiblePositions,
  ) as number[]);

{
  const constantFuel = identity;
  printSolution("part1", minFuel(constantFuel)); // 37, 337833
}

{
  const increasingFuel = (distance) => distance * (distance + 1) / 2;
  printSolution("part2", minFuel(increasingFuel)); // 168, 96678050
}
